src/common/utils/web3/createProvider.ts

```diff
@@ -1,8 +1,18 @@
 import { ethers } from "ethers";
 import { infuraId } from "common/config";
 
-const provider = new ethers.providers.JsonRpcProvider(
+let provider = new ethers.providers.JsonRpcProvider(
   `https://mainnet.infura.io/v3/${infuraId}`
 );
 
+if (process.env.REACT_APP_CURRENT_ENV === "kovan") {
+  provider = new ethers.providers.JsonRpcProvider(
+    `https://kovan.infura.io/v3/${infuraId}`
+  );
+}
+
+if (process.env.REACT_APP_CURRENT_ENV === "test") {
```

Comment:
would be great if this was captured in a function, rather than using envs outside of the config
```js

function Provider(networkId){
  const config = Config.get(networkId)
  return new ethers.providers.JsonRPCProvider(config.providerUrl)
}
```

or something like that, wdyt?